Sources/PluginLibrary/ProtoLanguage.swift

```diff
@@ -0,0 +1,42 @@
+// Sources/PluginLibrary/ProtoLanguage.swift - Proto language utilities
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// Utility functions for dealing with proto language syntax
+///
+// -----------------------------------------------------------------------------
+
+import Swift
+import Foundation
+
+public func isValidProtobufIdentifier(_ s: String) -> Bool {
```

Comment:
Is this used?

btw-I'm mixed on some of these validations.  For them to get to the plugin, protoc has to have parsed and approved them already.  So a lot of our validation doesn't really have to handle everything Swift can do as much as just handle what is valid proto but not valid Swift.